roles/fio-distributed/tasks/main.yml

```diff
@@ -17,6 +17,30 @@
     fio_path: /tmp
   when: fiod.storageclass is not defined
 
+- name: Download custom fio job
+  get_url:
+    url: "{{ fiod.job_url }}"
+    validate_certs: "{{ fiod.job_validate_certs }}"
+    dest: /tmp/fio.job
+  when: fiod.job_url is defined and fiod.job_url != ""
+
+- name: Generate fio test
+  k8s:
+    definition:
+      apiVersion: v1
+      kind: ConfigMap
+      metadata:
+        name: fio-test
+        namespace: '{{ operator_namespace }}'
+      data:
+        fiojob: "{{ lookup('file', '/tmp/fio.job') }}"
+  when: fiod.job_url is defined and fiod.job_url != ""
+
+- name: Delete fio job file
```

Comment:
Is there a need to delete the job file, since once the container has ended the job , the container has to be deleted, that means all the associated data in it too, unless the data is added onto a persistent volume, which in this case not going to happen.

Reply:
The ansible code is executed in the ripsaw operator itself, this pod is not ephemeral so that we need to remove the job file after.